src/main/scala/no/ndla/articleapi/repository/ArticleRepository.scala

```diff
@@ -28,53 +28,65 @@ trait ArticleRepository {
       val dataObject = new PGobject()
       dataObject.setType("jsonb")
       dataObject.setValue(write(article))
 
-      val count = sql"update ${Article.table} set document=${dataObject} where id=${article.id}".update().apply
+      val newRevision = article.revision.getOrElse(0) + 1
+      val count = sql"update ${Article.table} set document=${dataObject}, revision=$newRevision where id=${article.id} and revision=${article.revision}".update.apply
+
       if (count != 1) {
-        val message = s"Could not find article with id ${article.id}"
+        val message = s"Found revision mismatch when attempting to update article ${article.id}"
         logger.info(message)
-        Failure(NotFoundException(message))
+        Failure(new OptimisticLockException)
       } else {
         logger.info(s"Updated article ${article.id}")
-        Success(article)
+        Success(article.copy(revision=Some(newRevision)))
       }
     }
 
     def insertWithExternalIds(article: Article, externalId: String, externalSubjectId: Seq[String])(implicit session: DBSession = AutoSession): Long = {
+      val startRevision = 1
       val dataObject = new PGobject()
       dataObject.setType("jsonb")
       dataObject.setValue(write(article))
 
-      val articleId: Long = sql"insert into ${Article.table} (external_id, external_subject_id, document) values (${externalId}, ARRAY[${externalSubjectId}], ${dataObject})".updateAndReturnGeneratedKey().apply
+      val articleId: Long = sql"insert into ${Article.table} (external_id, external_subject_id, document, revision) values (${externalId}, ARRAY[${externalSubjectId}], ${dataObject}, $startRevision)".updateAndReturnGeneratedKey().apply
 
       logger.info(s"Inserted node $externalId: $articleId")
       articleId
     }
 
-    def updateWithExternalId(article: Article, externalId: String)(implicit session: DBSession = AutoSession): Long = {
+    def updateWithExternalId(article: Article, externalId: String)(implicit session: DBSession = AutoSession): Try[Long] = {
       val dataObject = new PGobject()
       dataObject.setType("jsonb")
       dataObject.setValue(write(article))
 
-      val articleId: Long = sql"update ${Article.table} set document=${dataObject} where external_id=${externalId}".updateAndReturnGeneratedKey().apply
-
-      logger.info(s"Updated node $externalId: $articleId")
-      articleId
+      val revision = article.revision.getOrElse(0) + 1
+      Try(sql"update ${Article.table} set document=${dataObject} where external_id=${externalId} and revision=$revision".updateAndReturnGeneratedKey().apply) match {
+        case Success(articleId) => {
+          logger.info(s"Updated article with external_id=$externalId, id=$articleId")
+          Success(articleId)
+        }
+        case Failure(ex) => {
+          val message = s"Found revision mismatch when attempting to update article with external id $externalId. The article has been edited on the new NDLA platform"
```

Comment:
Du vet vel ikke hvordan artikkelen har blitt editert? Bare at den har blitt det?
Ville endret teksten til noe mer "løsningsorientert". F.eks.:
`The revision stored in the database is newer than the one being updated. Please use the latest version from database when updating.`